packages/dev/core/src/Engines/thinEngine.ts

```diff
@@ -1790,8 +1790,20 @@ export class ThinEngine {
         }
 
         if (IsWindowObjectExist()) {
-            width = this._renderingCanvas ? this._renderingCanvas.clientWidth || this._renderingCanvas.width : window.innerWidth;
-            height = this._renderingCanvas ? this._renderingCanvas.clientHeight || this._renderingCanvas.height : window.innerHeight;
+            if (this._renderingCanvas) {
+                const boundingRect = this._renderingCanvas.getBoundingClientRect
+                    ? this._renderingCanvas.getBoundingClientRect()
+                    : {

```

Comment:
should we prevent the GC here ?

Reply:
GC? you mean the creation of this object?
This object will not be created unless getBoundingClientRect  doesn't exist, which is only the case in very (very) old browsers. Adding this object to the engine as a private member will only make the thin engine consume a little more memory for the chance of saving a single GC call on resize call.